### ISSUE TITLE
 Configure DynamoDB endpoints in AWS.

### ISSUE THREAD

**Issue by moisesmejia1200** [gravitational/teleport#13222]
What would you like Teleport to do?

Customer would like to make their environment use FIPS 140-2 compliant and use Teleport in FIPS 140-2 mode. Customer would like the ability to configure DynamoDB endpoints in AWS

What problem does this solve?
The customer would like to have Teleport run in FIPS mode to use AWS FIPS endpoints to protect the Teleport state backend. 

If a workaround exists, please include it.
No workaround exist at the moment 

Customer is on version: 6.2 but expect to be more current, probably up to at least 8.x, .

**Comment by zmb3** [comment]
Looks like a duplicate of #11291

**Comment by MarcClusterman** [comment]
> #11291

Looks similar, but that issue appears limited to S3 endpoints, while this one concerns DynamoDB endpoints which do not use virtual hosted-style addressing. Goal here is to configure to connect to `dynamodb-fips.us-east-1.amazonaws.com` (or whatever region) rather than connecting to `dynamodb.us-east-1.amazonaws.com`, which it doesn't appear to do now.

**Comment by r0mant** [comment]
@MarcClusterman Yes, `AWS_USE_FIPS_ENDPOINT` should work as DynamoDB client supports the same endpoint resolver as S3 client:

https://github.com/aws/aws-sdk-go-v2/blob/service/dynamodb/v1.15.3/service/dynamodb/internal/endpoints/endpoints.go#L35

We will still add a better support for this to Teleport to allow toggling FIPS mode without having to set the env. var but let us know if this workaround unblocks you for the time being. Thanks!

**Comment by MarcClusterman** [comment]
Unfortunately using `AWS_USE_FIPS_ENDPOINT=true` doesn't work in AWS East/West regions because it causes the service to attempt to also use it for DynamoDB streams, but AWS doesn't have a FIPS endpoint for streams there (only in GovCloud) apparently. This isn't really a Teleport issue, it's an AWS limitation and an issue with their SDK that interprets the environment variable.

**Comment by alistanis** [comment]
@MarcClusterman In that case I'd recommend using the setting that this issue introduced, as using that will not cause the AWS SDK to attempt to use any other FIPS endpoints for any other services. Unmerged docs PR here to see how to configure the Teleport config: https://github.com/gravitational/teleport/pull/14184

As of now I'm closing this issue due to the fact that, as you mention, anything further here really seems like an AWS limitation. If that doesn't end up being the case please feel free to let me know. Thanks!
